tests/exchange_online/test_binance_compare_ohlcv.py

```diff
@@ -0,0 +1,102 @@
+ARCHIVE: 2019-12-31 00:00:00 │ 2020-02-29 23:59:00 │   87840
+API:     2019-12-19 08:57:00 │ 2020-03-01 06:55:00 │  104999
+"""
+
+import asyncio
+import os
+
+import aiohttp
+import pytest
+
+from freqtrade.exchange.binance_public_data import binance_vision_zip_url
+from freqtrade.util.datetime_helpers import dt_from_ts
+
+
+class Check:
+    def __init__(self, asset_type_url_segment, timeframe):
+        self.asset_type_url_segment = asset_type_url_segment
+        self.timeframe = timeframe
+        self.klines_endpoint = "https://api.binance.com/api/v3/klines"
+        self.exchange_endpoint = "https://api.binance.com/api/v3/exchangeInfo"
+        self.mismatch = set()
+
+        if asset_type_url_segment == "futures/um":
+            self.klines_endpoint = "https://fapi.binance.com/fapi/v1/klines"
+            self.exchange_endpoint = "https://fapi.binance.com/fapi/v1/exchangeInfo"
+
+    async def check_one_symbol(self, symbol):
+        async with self.session.get(
+            self.klines_endpoint, params=dict(symbol=symbol, interval=self.timeframe, startTime=0)
+        ) as resp:
+            resp.raise_for_status()
+            json = await resp.json()
+            first_kline = json[0]
+            first_kline_ts = first_kline[0]
+            date = dt_from_ts(first_kline_ts).date()
+
+        archive_url = binance_vision_zip_url(
+            self.asset_type_url_segment, symbol=symbol, timeframe=self.timeframe, date=date
+        )
+        async with self.session.get(
+            archive_url, params=dict(symbol=symbol, interval=self.timeframe, startTime=0)
+        ) as resp:
+            if resp.status != 200:
+                self.mismatch.add(symbol)
+                print(
+                    f"{resp.status} API first kline: {dt_from_ts(first_kline_ts).isoformat()}  "
+                    f"{archive_url}"
+                )
+                web_url = archive_url.rsplit("/", 1)[0].replace(
+                    "https://data.binance.vision/", "https://data.binance.vision/?prefix="
+                )
+                print(f"Check {web_url}")
+
+    async def get_symbols(self):
+        async with self.session.get(self.exchange_endpoint) as resp:
+            resp.raise_for_status()
+            json = await resp.json()
+            symbols = [
+                symbol["symbol"]
+                for symbol in json["symbols"]
+                if not symbol["status"] == "PENDING_TRADING"
+            ]
+            return symbols
+
+    async def run(self) -> list:
+        async with aiohttp.ClientSession() as session:
+            self.session = session
+            symbols = await self.get_symbols()
+            await asyncio.gather(*[self.check_one_symbol(symbol) for symbol in symbols])
+            return self.mismatch
+
+
+@pytest.mark.skipif(
+    not bool(os.environ.get("TEST_BINANCE_COMPARE_OHLCV")),
+    reason="Simply to demonstrate the availabity of the archive endpoint",
+)
+async def test_binance_compare_ohlcv():
```

Comment:
I'm torn on this whole file (if it's worth keeping, really).

it seems pretty useful per se - but it won't run - and is prone to fail (it currently fails on 3 new pairs).
On the other hand - it only tests "is the pair available" - not "is the full timerange available" - which is where i suspect more problems / need for future investigation.

Reply:
this file is like the previously removed one, it gives us some confidence that not too many pairs will fail at first.

Reply:
> I guess the result is relatively stable.

it's not unfortunately - not without further filters
I've ran it yesterday evening, and it failed on 3 more pairs (6 in total) - all pairs (not in your list above) that have only been listed within the last day or two.

As binance lists pairs all the time - it'd be a test we expect to fail one day a week at least ... if not more 😆